src/test/ui/pub/pub-reexport-priv-extern-crate.stderr

```diff
@@ -1,30 +1,36 @@
-error: extern crate `core` is private, and cannot be re-exported (error E0365), consider declaring with `pub`
-  --> $DIR/pub-reexport-priv-extern-crate.rs:4:9
+error[E0365]: `core` is private, and cannot be re-exported
+  --> $DIR/pub-reexport-priv-extern-crate.rs:2:9
    |
 LL | pub use core as reexported_core;
-   |         ^^^^^^^^^^^^^^^^^^^^^^^
+   |         ^^^^^^^^^^^^^^^^^^^^^^^ re-export of private `core`
    |
-   = note: `#[deny(pub_use_of_private_extern_crate)]` on by default
-   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
-   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>
+   = note: consider declaring type or module `core` with `pub`
 
-error: extern crate `core` is private, and cannot be re-exported (error E0365), consider declaring with `pub`
-  --> $DIR/pub-reexport-priv-extern-crate.rs:12:9
+error[E0603]: crate import `core` is private
+  --> $DIR/pub-reexport-priv-extern-crate.rs:9:15
    |
 LL |     use foo1::core;
-   |         ^^^^^^^^^^
+   |               ^^^^ private crate import
    |
-   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
-   = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>
+note: the crate import `core` is defined here
+  --> $DIR/pub-reexport-priv-extern-crate.rs:5:5
+   |
+LL |     extern crate core;
+   |     ^^^^^^^^^^^^^^^^^^
```

Comment:
Looking at these with fresh eyes, they could easily be structured suggestions, right?

Reply:
Not sure what do you mean, this is a `span_note`.

(Anyway, changing diagnostics here would result in a large diff, and I don't want to make orthogonal changes to diagnostics in the same PR with changing the language, same applies to https://github.com/rust-lang/rust/pull/80763/files#r555486463)